pkg/sql/lease.go

```diff
@@ -697,20 +706,30 @@ func (t *tableState) findForTimestamp(
 //
 // t.mu must be locked.
 func (t *tableState) acquireFreshestFromStoreLocked(ctx context.Context, m *LeaseManager) error {
-	// Ensure there is no lease acquisition in progress.
-	t.acquireWait()
-
-	// Move forward to acquire a fresh table lease.
+	// Continue to attempt to call to acquireNodeLease until our call was the one
+	// made by singleflight. This busy-waits while another acquisitions are
+	// underway.
+	var err error
+	hasAcquiredFreshest := false
+	for !hasAcquiredFreshest {
```

Comment:
```for hasAcquiredFreshest := false; !hasAcquiredFreshest; {```

Reply:
maybe not. See my comment below. Just break out of this loop once shared == false